controller/array_action/array_mediator_interface.py

```diff
@@ -179,6 +202,19 @@ def get_snapshot(self, snapshot_name):
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def get_snapshot_by_id(self, snapshot_id):
+        """
+        This function return snapshot info about the snapshot.
+        Args:
+            snapshot_id : id of the snapshot in the storage system
+        Returns:
+           Snapshot
+        Raises:
+            SnapshotNameBelongsToVolumeError
```

Comment:
but in other storage systems snapshot IS a volume, how can you tell?

Reply:
In a9k too but volume/snapshot has field master_name which is volume name it is created from.
If this value is empty it is volume, if not - snapshot
I assume other storages also has a way to check. Anyway in this ticket i couldn't consider all other storages.